src/main/kotlin/DirectoryLogic.kt

```diff
@@ -2,20 +2,57 @@ import java.lang.IllegalArgumentException
 
 class DirectoryLogic {
 
-    private val supportedExtensions = arrayOf("wav", "mp3", "jpeg", "tr")
+    private val supportedExtensions = arrayOf("wav", "mp3", "jpeg", "jpg", "tr")
+    private val supportedContainers = arrayOf("tr")
+    private val compressedTypes = arrayOf("mp3", "jpeg", "jpg")
+    private val groupings = arrayOf("book", "chapter", "verse", "chunk")
 
+    @Throws(IllegalArgumentException::class)
     fun buildFullFilePath(
-        inputFilePath: String,
-        languageCode: String,
-        dublinCoreId: String,
-        group: String,
-        projectId: String = "",
-        mediaExtension: String = "",
-        mediaQuality: String = "hi"
+            inputFilePath: String,
+            languageCode: String,
+            dublinCoreId: String,
+            group: String,
+            projectId: String = "",
+            mediaExtension: String = "",
+            mediaQuality: String = ""
     ): String {
 
-        return ""
+        val fileExt = inputFilePath.split("/").last().split('.').last()
```

Comment:
Splitting on / isn't really useful, not all operating systems will use / (windows can use \, as seen below in your test cases)

splitting on . and getting the last will work in that case, but would even if you didn't split on /

Though Iif you have inputFile as a File object though you can just do:
```
inputFile.extension
```